projects/concurrency/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"container/list"
+	"sync"
+)
+
+type Cache[K comparable, V any] struct {
+	capacity int
+	entries  map[K]*list.Element
+	list     *list.List
+	mu       sync.Mutex
+}
+
+type CacheValue[K comparable, V any] struct {
+	// i don't like that i am duplicating the key here... but it is for easier lookup later
+	key   K
+	value V
+}
+
+func NewCache[K comparable, V any](entryLimit int) Cache[K, V] {
```

Comment:
What happens if you pass 0 or -1 as the `entryLimit` and then try to use the cache?